File: backend/src/main/java/com/quizmatrix/quizmatrix/model/Question.java
package com.quizmatrix.quizmatrix.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity(name = "Question")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Question {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Integer id_question;

    @OneToMany(mappedBy ="question", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Answer> answers;

    @ManyToOne
    @JoinColumn(name = "id_quiz")
    private Quiz quiz;
    private String text;
    private Integer points;
}

File: frontend/QuizMatrix/src/app/models/question.ts
import { Answer } from './answer';

export class Question {
  id_question: number;
  id_quiz: number;
  text: string;
  points: number;
  answers: Answer[] = [];

  constructor(
    id_question: number,
    id_quiz: number,
    text: string,
    points: number
    // answer: Answer[]
  ) {
    this.id_question = id_question;
    this.id_quiz = id_quiz;
    this.text = text;
    this.points = points;
    //  this.answers = answer;
  }
}
